backend-tests/tests/test_deployments.py

```diff
@@ -1446,3 +1450,613 @@ class TestDeploymentsStatusUpdateEnterprise(TestDeploymentsStatusUpdateBase):
+def clean_mongo_client(mongo):
+    """Fixture setting up a clean (i.e. empty database). Yields
+    common.MongoClient connected to the DB.
+
+    Useful for tests with multiple testcases:
+    - protects the whole test func as usual 
+    - but also allows calling MongoClient.cleanup() between cases
+    """
+    mongo_cleanup(mongo)
+    yield mongo
+    mongo_cleanup(mongo)
+
+
+def predicate(attr, scope, t, val):
+    return {"attribute": attr, "scope": scope, "type": t, "value": val}
+
+
+def create_filter(name, predicates, utoken):
+    f = {"name": name, "terms": predicates}
+
+    r = (
+        ApiClient(inventory_v2.URL_MGMT)
+        .with_auth(utoken)
+        .call("POST", inventory_v2.URL_FILTERS, f)
+    )
+    assert r.status_code == 201
+
+    f["id"] = r.headers["Location"].split("/")[1]
+    return f
+
+
+def create_dynamic_deployment(
+    name, predicates, utoken, max_devices=None, phases=None, status_code=201
+):
+    f = create_filter(name, predicates, utoken)
+
+    api_dep_v2 = ApiClient(deployments_v2.URL_MGMT)
+    api_dep_v1 = ApiClient(deployments.URL_MGMT)
+
+    depid = None
+    with get_mender_artifact(name) as filename:
+
+        upload_image(filename, utoken)
+
+        deployment_req = {
+            "name": name,
+            "artifact_name": name,
+            "filter_id": f["id"],
+        }
+
+        if max_devices is not None:
+            deployment_req["max_devices"] = max_devices
+
+        if phases is not None:
+            deployment_req["phases"] = phases
+
+        res = api_dep_v2.with_auth(utoken).call(
+            "POST", deployments_v2.URL_DEPLOYMENTS, deployment_req
+        )
+
+        assert res.status_code == status_code
+        if status_code != 201:
+            return None
+
+        depid = res.headers["Location"].split("/")[5]
+
+    newdep = get_deployment(depid, utoken)
+
+    assert newdep["name"] == name
+    assert newdep["filter"]["id"] == f["id"]
+    assert newdep["filter"]["terms"] == predicates
+    assert newdep["status"] == "pending"
+    assert newdep["dynamic"]
+
+    return newdep
+
+
+def get_deployment(depid, utoken):
+    api_dep_v1 = ApiClient(deployments.URL_MGMT)
+    res = api_dep_v1.with_auth(utoken).call(
+        "GET", deployments.URL_DEPLOYMENTS_ID, path_params={"id": depid}
+    )
+    assert res.status_code == 200
+    return res.json()
+
+
+def make_device_with_inventory(attributes, utoken, tenant_token):
+    devauthm = ApiClient(deviceauth_v2.URL_MGMT)
+    devauthd = ApiClient(deviceauth_v1.URL_DEVICES)
+
+    d = make_accepted_device(utoken, devauthd, tenant_token)
+
+    submit_inventory(attributes, d.token)
+
+    d.attributes = attributes
+
+    return d
+
+
+def submit_inventory(attrs, token):
+    invd = ApiClient(inventory.URL_DEV)
+    r = invd.with_auth(token).call("PATCH", inventory.URL_DEVICE_ATTRIBUTES, attrs)
+    assert r.status_code == 200
+
+
+def update_deployment_status(deployment_id, status, token):
```

Comment:
if I understand correctly this should be more like update_device_deployment_status(device_deployment_id,...
in the deployments service, we have deployments and device deployments and it would be good not to mix those things here